core/test/regression/api/canary/admin/ghost_identity_spec.js

```diff
@@ -0,0 +1,102 @@
+const should = require('should');
+const supertest = require('supertest');
+const jwt = require('jsonwebtoken');
+const jwksClient = require('jwks-rsa');
+const testUtils = require('../../../../utils');
+const localUtils = require('./utils');
+const config = require('../../../../../server/config');
+
+const ghost = testUtils.startGhost;
+
+let request;
+
+const verifyJWKS = (endpoint, token) => {
+    return new Promise((resolve, reject) => {
+        const jwksClient = require('jwks-rsa');
+        const client = jwksClient({
+            jwksUri: endpoint
+        });
+
+        function getKey(header, callback){
+            client.getSigningKey(header.kid, (err, key) => {
+                let signingKey = key.publicKey || key.rsaPublicKey;
+                callback(null, signingKey);
+            });
+        }
+
+        jwt.verify(token, getKey, {}, (err, decoded) => {
+            if (err) {
+                reject(err);
+            }
+
+            resolve(decoded);
+        });
+    });
+};
+
+describe('Ghost Identity API', function () {
+    describe('As Owner', function () {
+        before(function () {
+            return ghost()
+                .then(function () {
+                    request = supertest.agent(config.get('url'));
+                })
+                .then(function () {
+                    return localUtils.doAuth(request);
+                });
+        });
+
+        it('Can create JWT token and verify it afterwards with public jwks', function () {
+            let identity;
+
+            return request
+                .get(localUtils.API.getApiQuery(`identity/`))
+                .set('Origin', config.get('url'))
+                .expect('Content-Type', /json/)
+                .expect('Cache-Control', testUtils.cacheRules.private)
+                .expect(200)
+                .then((res) => {
+                    should.not.exist(res.headers['x-cache-invalidate']);
+                    const jsonResponse = res.body;
+                    should.exist(jsonResponse);
+
+                    identity = jsonResponse;
+                })
+                .then(() => {
+                    return verifyJWKS(`${request.app}/ghost/.well-known/jwks.json`, identity);
```

Comment:
@allouis used this lib to verify the token as it seemed the closest to real world verification scenario and also was described as a part of [jsonwebtoken documentation](https://github.com/auth0/node-jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback)